service/create/security_group.go

```diff
@@ -71,30 +71,34 @@ func (s *Service) deleteSecurityGroup(input securityGroupInput) error {
 }
 
 // masterRules returns the rules for the masters security group.
-func (ri rulesInput) masterRules() []awsresources.Rule {
-	return []awsresources.Rule{
+func (ri rulesInput) masterRules() []awsresources.SecurityGroupRule {
+	return []awsresources.SecurityGroupRule{
 		{
 			Port:       ri.Cluster.Spec.Cluster.Kubernetes.API.SecurePort,
 			SourceCIDR: defaultCIDR,
 		},
 		{
 			Port:            ri.Cluster.Spec.Cluster.Etcd.Port,
-			SecurityGroupID: ri.OwnSecurityGroupID,
+			SecurityGroupID: ri.MastersSecurityGroupID,
```

Comment:
Didn't we say we need this because we connect to etcd via the LB, and not `localhost`?